libs/blocks/merch-card/merch-card.js

```diff
@@ -168,17 +162,60 @@ const init = (el) => {
       }
     }
   });
-  addWrapper(el, section, cardType);
-  merchCard.classList.add('consonant-Card', 'consonant-ProductCard', `consonant-${cardType}`);
-  if (image) addBackgroundImg(image, cardType, merchCard);
-  if (ribbonMetadata !== null) decorateRibbon(el, ribbonMetadata, cardType);
-  image?.parentElement.remove();
-  if (ctas) decorateButtons(ctas);
-  addInner(el, altCta, cardType, merchCard);
-  decorateIcon(el, icons, cardType);
-  const inner = el.querySelector(`.consonant-${cardType}-inner`);
-  const innerCleanup = inner.querySelectorAll(':scope > div')[1];
-  if (innerCleanup.classList.length === 0) innerCleanup.remove();
+
+  const merchCard = createTag('merch-card', { class: el.className, variant: cardType });
+  if (ribbonMetadata !== null) {
+    const badge = returnRibbonStyle(ribbonMetadata);
+    if (badge !== null) {
+      merchCard.setAttribute('badge-background-color', badge.badgeBackgroundColor);
+      merchCard.setAttribute('badge-color', badge.badgeColor);
+      merchCard.setAttribute('badge-text', badge.badgeText);
+    }
+  }
+  if (actionMenuContent !== null) {
+    merchCard.setAttribute('action-menu', true);
+    merchCard.append(createTag('div', { slot: 'action-menu-content' }, actionMenuContent.innerHTML));
+  }
+  if (ctas) {
+    const footer = createTag('div', { slot: 'footer' });
+    decorateButtons(ctas);
+    footer.append(ctas);
+    merchCard.appendChild(footer);
+  }
+  if (image !== undefined) {
+    merchCard.setAttribute('image', image.querySelector('img').src);
+    image.remove();
+  }
+  if (!icons || icons.length > 0) {
+    const iconImgs = Array.from(icons).map((icon) => {
+      const img = {
+        src: icon.querySelector('img').src,
+        alt: icon.querySelector('img').alt,
+      };
+      return img;
+    });
+    merchCard.setAttribute('icons', JSON.stringify(Array.from(iconImgs)));
+    icons.forEach((icon) => icon.remove());
+  }
+  if (styles.includes('secure')) {
+    replaceKey('secure-transaction', getConfig())
+      .then((key) => merchCard.setAttribute('secure-label', key));
+  }
+  if (altCta) {
+    const label = checkBoxLabel(ctas, altCta);
+    if (label !== null) {
+      merchCard.setAttribute('checkbox-label', label);
+    }
+  }
```

Comment:
Do a lot of checks in the instantiation, can de nest here too I think

```suggestion
  const label = checkBoxLabel(ctas, altCta);
  if (label !== null) {
    merchCard.setAttribute('checkbox-label', label);
  }
```